### PR TITLE
Require Web UI Password Only for POST Requests or Sensitive Information

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

Passwords for our web UI today lead to a poor user experience, as they are always needed when reloading the page. Instead, we should require a password when the user is performing a sensitive action such as importing keystores or creating a wallet, and then not require them for read-only data. All the data in the web UI is accessible in block explorers and is not sensitive.

The auth flow remains similar, except that token expirations are reduced to a minute. When users are performing a sensitive action, they will login via a pop-up in the web UI to receive a JWT token, then their post request will include the bearer token in its http headers. The token will then be thrown away by the web UI.

**Which issues(s) does this PR fix?**

Part of https://github.com/prysmaticlabs/prysm-web-ui/issues/137

### CODE REVIEW

File: validator/slashing-protection/cli_export.go [hunk 1]
```diff
@@ -2,6 +2,8 @@ package slashingprotection
 
 import (
 	"encoding/json"
+	"path/filepath"
+
```
Comment (rauljordan) [inline validator/slashing-protection/cli_export.go:6]:
```suggestion
```

**Review by rkapka — APPROVED** [review]

**Comment by rkapka** [comment]
I approved the PR, but I have one question: if we merge this, it will break the client, right? If I'm not mistaken, we need to ship UI v1 at the same time as this change because one cannot work properly without the other.

**Comment by rauljordan** [comment]
Agreed and will work on the web changes first so we do not break compatibility

**Comment by rauljordan** [comment]
Ended up consolidating this into a feature branch instead, as there are future PRs that will depend on this one and much easier to track a webv1 branch instead

File: validator/rpc/auth_test.go [hunk 2]
```diff
@@ -74,28 +72,6 @@ func TestServer_SignupAndLogin_RoundTrip(t *testing.T) {
...
-		return ss.jwtKey, nil
-	}
-	_, err = jwt.Parse(tokenString, checkParsedKey)
-	assert.NoError(t, err)
-
-	_, err = ss.Logout(context.Background(), &ptypes.Empty{})
-	require.NoError(t, err)
-
-	// Attempting to validate the same token string after logout should fail.
-	_, err = jwt.Parse(tokenString, checkParsedKey)
```
Comment (0xKiwi) [inline validator/rpc/auth_test.go:95]: Is there still a test for the token string?

File: validator/rpc/intercepter_test.go [hunk 3]
```diff
@@ -40,8 +40,13 @@ func TestServer_JWTInterceptor_BadToken(t *testing.T) {
 	}
 	interceptor := s.JWTInterceptor()
 
+	var methodPath string
+	for k := range authPaths {
+		methodPath = k
+		break
+	}
```
Comment (0xKiwi) [inline validator/rpc/intercepter_test.go:47]: What is this supposed to be doing?

**Review by 0xKiwi — APPROVED** [review]

**Pull request merged**